refactor: drop need for an obscure (and legacy) `OC_Util` method... `runningOnMac()` ;-)

joshtrichards opened the pull request:
<!--
  - 🚨 SECURITY INFO
  -
  - Before sending a pull request that fixes a security issue please report it via our HackerOne page (https://hackerone.com/nextcloud) following our security policy (https://nextcloud.com/security/). This allows us to coordinate the fix and release without potentially exposing all Nextcloud servers and users in the meantime.
-->

## Summary

- Standard PHP constant available
- One less legacy (OC_Util) function[^deprecated]
- Easier to read
- Nobody was using it anyway
- Technically more robust

[^deprecated]: Once we feel comfortable removing it. Not really sure if it's considered a public API or not. I guess kinda sorta. So `@deprecated` it (and changed its implementation to use the constant meanwhile too).

## TODO

- [ ] ...

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)
- [ ] [Labels added](https://github.com/nextcloud/server/labels) where applicable (ex: bug/enhancement, `3. to review`, feature component)
- [ ] [Milestone added](https://github.com/nextcloud/server/milestones) for target branch/version (ex: 32.x for `stable32`)

joshtrichards committed:
chore(legacy): deprecate OC_Util::runningOnMac()

- Easy enough query directly these days; Only used in one spot anyhow
- Updated the runningOnMac method to use PHP_OS_FAMILY just for clarity until we can remove the function entirely (same result; supported since PHP ~7.4).

Signed-off-by: Josh <josh.t.richards@gmail.com>

joshtrichards committed:
refactor(tests): use OS constant over function

Use PHP's newer PHP_OS_FAMILY constant.

- Allows us to drop a legacy OC_Util method.
- This test isn't even enabled at the moment.

Signed-off-by: Josh <josh.t.richards@gmail.com>

joshtrichards committed:
refactor(setup): use OS constant rather than function (macOS)

PHP_OS_FAMILY

- Facilitates elimination of a legacy OC_Util method
  - `runningOnMac()` -- yes, really! ;-)
- Supported since PHP late 7.x-era

Signed-off-by: Josh <josh.t.richards@gmail.com>

CarlSchwan reviewed (approved):
One less :)

CarlSchwan committed:
fix(formatting): Fix formatting issue

Signed-off-by: Carl Schwan <carl.schwan@nextcloud.com>

szaimen reviewed (approved):
LGTM but did not test

come-nc commented on lib/private/legacy/OC_Util.php:
```diff
@@ -733,12 +733,16 @@ public static function obEnd() {
 	 * @return bool true if running on Mac OS X, false otherwise
+	 * @deprecated 33.0.0 Query PHP_OS_FAMILY directly.
 	 */
 	public static function runningOnMac() {
```
`OC_*` is a bit of a grey area, it used to be public API a long time ago so for some methods we still use the deprecation process.
For this one it’s quite specific and unused so it can safely be removed I think.

joshtrichards committed:
chore: drop no longer needed runningOnMac method

Removed deprecated method 'runningOnMac' that checks if PHP is running on macOS.

Signed-off-by: Josh <josh.t.richards@gmail.com>

joshtrichards committed:
chore: drop unused Darwin range test variation

Signed-off-by: Josh <josh.t.richards@gmail.com>

SystemKeeper commented:
This breaks MacOS CI as the error is now not ignored anymore on install: https://github.com/nextcloud/talk-ios/actions/runs/20001229346/job/57357971111?pr=2306

Needs adjustment at:
https://github.com/nextcloud/server/blob/bbca4fe56e72a49e1380d2f9630a48faffe9e8cc/core/Command/Maintenance/Install.php#L64
(Leaving out the question if there’s a better way to do this ..)

provokateurin commented:
Since we don't support macOS, you should also not use it in your CI this way :see_no_evil: The right way is probably using a docker container for this in CI
